### ISSUE TITLE
strictNullCheck with Dictionary `Index signature is missing in type '{}'.`

### ISSUE THREAD

**Issue by alan-agius4** [Microsoft/TypeScript#19464]
I am having some issues with Dictionaries and StrictNullChecks and I believe that I spotted a bug

**TypeScript Version:**  2.5.2

**Code**

```ts
interface Dictionary {
	[key: string]: string;
}

const y: Dictionary = {}; // TypeScript doesn't complain that this Index signature is missing

let x: Dictionary | undefined;
const z: Dictionary = { ...x }; // Typescript complains that Index signature is missing type is {} | Dictionary (rightly so)

```

**Expected behavior:**
Errors to be consistent in both or in none

**Actual behavior:**
Not consistent

**Comment by mhegazy** [comment]
Seems like a duplicate of https://github.com/Microsoft/TypeScript/issues/16694. should be fixed in `typescript@next`
